temporal-sdk/src/main/java/io/temporal/client/BuildIDOperation.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client;
+
+import io.temporal.api.workflowservice.v1.UpdateWorkerBuildIdCompatibilityRequest;
+import javax.annotation.Nonnull;
+
+/**
+ * The implementations of this class can be passed as parameters to {@link
+ * WorkflowClient#updateWorkerBuildIDCompatability(String, BuildIDOperation)}
+ *
+ * <p>See each public static method to learn about and construct the available operations.
+ */
+public abstract class BuildIDOperation {
+  private BuildIDOperation() {}
+
+  abstract void augmentBuilder(UpdateWorkerBuildIdCompatibilityRequest.Builder builder);
+
+  /**
+   * This operation adds a new Build ID into a new set, which will be used as the default set for
+   * the queue. This means all new workflows will start on this Build ID.
+   *
+   * @param buildID The Build ID to add as the new overall default.
+   */
+  public static BuildIDOperation newIDInNewDefaultSet(@Nonnull String buildID) {
+    return new BuildIDOperation() {
```

Comment:
For legibility in stack traces and elsewhere, static inner classes are often clearer than anonymous classes (they don't add many more lines, don't need to be exposed (unless you prefer that over methods), etc)